noir/noir-repo/acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -382,6 +406,65 @@ impl<'a, B: BlackBoxFunctionSolver> ACVM<'a, B> {
         }
     }
 
+    fn extract_assertion_payload(
+        &self,
+        location: OpcodeLocation,
+    ) -> Option<ResolvedAssertionPayload> {
+        let found_assertion_payload = self.assertion_payloads.iter().find_map(|(loc, payload)| {
+            if location == *loc {
+                Some(payload)
+            } else {
+                None
+            }
+        });
+        let Some(found_assertion_payload) = found_assertion_payload else {
+            return None;
+        };
```

Comment:
```suggestion
        let Some(found_assertion_payload)  = self.assertion_payloads.iter().find_map(|(loc, payload)| {
            if location == *loc {
                Some(payload)
            } else {
                None
            }
        }) else {
            return None;
        };
```
looks like these can be combined

Reply:
```suggestion
        let (_, found_assertion_payload = self.assertion_payloads.iter().find(|(loc, _)| location == *loc)?;
```

Can be simplified further to

Reply:
Will do!